src/leader.py

```diff
@@ -142,14 +143,23 @@ def lead(self):
         Moves the leader vehicle based on controller input.
         """
         code, position = self._controller.get_input()
-        if code == 'right_trigger':
-            self.set_speed(position)
-            self.drive()
-        if code == 'left_trigger':
-            self.set_speed(position)
-            self.reverse()
-        if code == 'dpad_left_right' or code == 'left_stick_x':
-            self.turn(code, position)
+        if code is not None:
```

Comment:
It seems as soon as it calls the function once it will then call lead on the leader object again and return to manual control, we need some sort of way to keep it in a specific mode or calling a specific demo function.

Reply:
If the code is none then it won't switch to manual controls. Also on the release of a button, the code of the button is still called. I tested this and it works the way I think it should.